app/grandchallenge/algorithms/views.py

```diff
@@ -1013,3 +1022,52 @@ def get_success_message(self, cleaned_data):
 
     def get_success_url(self):
         return self.algorithm.get_absolute_url() + "#models"
+
+
+class AlgorithmImageTemplate(ObjectPermissionRequiredMixin, DetailView):
+    model = Algorithm
+    permission_required = "algorithms.change_algorithm"
+    raise_exception = True
+    queryset = Algorithm.objects.prefetch_related("inputs", "outputs")
+
+    def get(self, *_, **__):
+        algorithm = self.get_object()
+
+        forge_context = get_forge_algorithm_template_context(algorithm)
+
+        # Step 1: Create a temporary directory for generating content
+        with tempfile.TemporaryDirectory() as temp_dir:
+            # Step 2: Generate the content in the temp directory
+            algorithm_template_path = generate_algorithm_template(
+                context=forge_context,
+                output_path=Path(temp_dir),
+            )
+
+            # Step 3: Create an in-memory buffer to store the ZIP file
+            zip_buffer = io.BytesIO()
+
+            # Step 4: Create the ZIP archive in memory
+            with ZipFile(zip_buffer, "w") as zipf:
+                for foldername, _, filenames in os.walk(
+                    algorithm_template_path
```

Comment:
How are we certain that the `algorithm_template_path` is safe? We should only package files that are in `temp_dir`. We should double check that the file that is being included are actually in `temp_dir`.